### ISSUE TITLE
Mismatched typescript type for PublicKeyFormat in DeviceManagerClient.getDevice response

### ISSUE THREAD

**Issue by scottbez1** [googleapis/nodejs-iot#250]
When I request device info (via `DeviceManagerClient.getDevice`), the returned data includes information about public keys for that device. However, the "format" field of a public key credential, which is declared as a type of `google.cloud.iot.v1.PublicKeyFormat` (an enum with numeric values), is actually being returned as a string rather than the numeric value that's expected based on the type declaration.

Some example code:
```
import { v1 } from '@google-cloud/iot'

[...]

        const client = new v1.DeviceManagerClient()
        const [existingDevice] = await client.getDevice({ name /* Note: assume `name` is declared elsewhere with the full device path of the device I'm looking up */})
        if (exists(existingDevice.credentials)) {
            for (const credential of existingDevice.credentials) {
                if (exists(credential.publicKey)) {
                    console.log(credential.publicKey.format)
                }
            }
        }
```

My test iot device is configured with a `RS256_X509` public key. In the example above, `credential.publicKey.format` has a typescript type of `google.cloud.iot.v1.PublicKeyFormat` so the expected value to be logged above would be `1` (the numeric value of `google.cloud.iot.v1.PublicKeyFormat.RSA_X509_PEM`), but instead it unexpectedly logs the string `RSA_X509_PEM`.

The `getDevice` method is declared as returning the protobuf-generated type `protosTypes.google.cloud.iot.v1.IDevice` whereas it seems like the function actually returns a javascript object representation of that proto data where enums have been converted to their string values (which means it does not actually conform to the `protosTypes.google.cloud.iot.v1.IDevice` type specification, making the typescript types incorrect).

#### Environment details
  - `@google-cloud/iot` version: 1.6.1

**Comment by alexander-fenster** [comment]
Thank you @scottbez1. We are aware of this problem with enums and trying to figure out the best solution.

Until this is fixed, you can use `.fromObject` method to get the valid representation of the returned object:

```ts
import { v1 } from '@google-cloud/iot';
import * as protos from '@google-cloud/iot/build/protos/protos';

...
const [existingDeviceObj] = await client.getDevice({ name });
const existingDevice = protos.google.cloud.iot.v1.Device.fromObject(existingDeviceObj);
// existingDevice will have valid enum types
```

```

**Comment by alexander-fenster** [comment]
This should be fixed now! In v1.7.0 the types are updated to include `enum` string keys (`keyof typeof google.cloud.iot.v1.PublicKeyFormat`). This is the best we can do at this moment without breaking the world (we could configure `protobuf.js` to return actual enum values instead of strings, but JavaScript users won't appreciate that).
